components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/CORSManagementServiceComponent.java

```diff
@@ -50,8 +55,13 @@
     @Activate
     protected void activate(ComponentContext context) {
 
-        context.getBundleContext()
-                .registerService(CORSManagementService.class, new CORSManagementServiceImpl(), null);
+        BundleContext bundleContext = context.getBundleContext();
+        bundleContext.registerService(CORSOriginDAO.class,
+                new CORSOriginDAOImpl(), null);
+        bundleContext.registerService(CORSConfigurationDAO.class,
+                new CORSConfigurationDAOImpl(), null);
+
+        bundleContext.registerService(CORSManagementService.class, new CORSManagementServiceImpl(), null);
```

Comment:
Shall we add a try-catch block as done in [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/internal/RoleManagementServiceComponent.java#L42-L51). Otherwise it will be difficult to identify if something goes wrong during the activation.